control/proto/gateway.proto

```diff
@@ -215,6 +216,8 @@ message add_host_req {
 	string host_nqn = 2;
 	optional string psk = 3;
 	optional string dhchap_key = 4;
+	optional bytes encrypted_psk = 5;
+	optional bytes encrypted_dhchap_key = 6;
```

Comment:
Why do these need to be sent encrypted? The gRPC channel is secure, isn't it? So, if we want the secrets (DHCHAP and PSK) encrypted at rest, it should be the server doing it the encryption/decryption before writing/after reading the OMAP, no?

In fact, key encryption/decryption is usually are managed by the servers, while keys are sent "unencrypted" from clients, BUT over a secure channel (which provides it's own encryption). If both clients and servers know the encryption key, the encryption key becomes less secure.

My understanding is that this requirement is for "encryption-at-rest".

Having this only in the server side will also simplify the deployment, as Cephadm knows nothing about clients.

Reply:
@epuertat these are not sent by gRPC. The gRPC uses unencrypted value, then we change them to encrypted ones in the gateway before saving it to OMAP. As we use the same structures in OMAP as we do in gRPC we have it defined in the prtoto.